Makefile

```diff
@@ -136,19 +136,19 @@ CRD_WORKING_DIR := crd_work_dir
 # So we have to generate them together and then move them into place
 manifests: $(CONTROLLER_GEN) #EXHELP Generate WebhookConfiguration, ClusterRole, and CustomResourceDefinition objects.
 	mkdir $(CRD_WORKING_DIR)
-	$(CONTROLLER_GEN) crd paths="./api/v1/..." output:crd:artifacts:config=$(CRD_WORKING_DIR)
+	GOFLAGS="-tags=$(GO_BUILD_TAGS)" $(CONTROLLER_GEN) crd paths="./api/v1/..." output:crd:artifacts:config=$(CRD_WORKING_DIR)
 	mv $(CRD_WORKING_DIR)/olm.operatorframework.io_clusterextensions.yaml $(KUSTOMIZE_OPCON_CRDS_DIR)
 	mv $(CRD_WORKING_DIR)/olm.operatorframework.io_clustercatalogs.yaml $(KUSTOMIZE_CATD_CRDS_DIR)
 	rmdir $(CRD_WORKING_DIR)
 	# Generate the remaining operator-controller manifests
-	$(CONTROLLER_GEN) rbac:roleName=manager-role paths="./internal/operator-controller/..." output:rbac:artifacts:config=$(KUSTOMIZE_OPCON_RBAC_DIR)
+	GOFLAGS="-tags=$(GO_BUILD_TAGS)" $(CONTROLLER_GEN) rbac:roleName=manager-role paths="./internal/operator-controller/..." output:rbac:artifacts:config=$(KUSTOMIZE_OPCON_RBAC_DIR)
 	# Generate the remaining catalogd manifests
-	$(CONTROLLER_GEN) rbac:roleName=manager-role paths="./internal/catalogd/..." output:rbac:artifacts:config=$(KUSTOMIZE_CATD_RBAC_DIR)
-	$(CONTROLLER_GEN) webhook paths="./internal/catalogd/..." output:webhook:artifacts:config=$(KUSTOMIZE_CATD_WEBHOOKS_DIR)
+	GOFLAGS="-tags=$(GO_BUILD_TAGS)" $(CONTROLLER_GEN) rbac:roleName=manager-role paths="./internal/catalogd/..." output:rbac:artifacts:config=$(KUSTOMIZE_CATD_RBAC_DIR)
+	GOFLAGS="-tags=$(GO_BUILD_TAGS)" $(CONTROLLER_GEN) webhook paths="./internal/catalogd/..." output:webhook:artifacts:config=$(KUSTOMIZE_CATD_WEBHOOKS_DIR)
```

Comment:
It will not work :-( 
I am proposing we add its support here: https://github.com/kubernetes-sigs/controller-tools/issues/1158#issuecomment-2703818694

But see has no env var or flag that allow us to pass the build tag so far: https://github.com/kubernetes-sigs/controller-tools/blob/main/pkg/loader/loader.go#L373-L379

The best away so far is replace the dep to downgrade it, see: https://github.com/operator-framework/operator-controller/pull/1846

Reply:
I think downgrade is dangerous, and the best way would be to move things around instead (short of fixing controller-tools)